logging.js

```diff
@@ -197,7 +207,26 @@ export class LoggingClient {
 			.withError(error)
 			.withMessage(developerMessage)
 			.withLocation()
-			.build());
-		this._logger.logBatch(logs);
+			.build()
+		).filter(this._throttle.bind(this));
+		if (logs.length > 0) {
+			this._logger.logBatch(logs);
+		}
+	}
+
+	_throttle(log) {
+		if (!this._shouldThrottle) {
+			return true;
+		}
+
+		const now = new Date().getTime();
```

Comment:
So now we _are_ calculating the time on every loop though the log array, because this function is now _inside_ `filter`.  I'm not super concerned though - the lms doesn't use the batch functions other than for errors thrown before the logger exists.  After that, it always uses the `legacyError` function, creating an array of size one, so optimizations wouldn't be much good.